internal/controller/stages/stages.go

```diff
@@ -647,198 +652,196 @@ func (r *reconciler) syncNormalStage(
 	status.ObservedGeneration = stage.Generation
 	status.Health = nil
 
-	if status.CurrentFreight == nil {
+	if currentFreight := status.FreightHistory.Current(); currentFreight == nil || len(currentFreight.Items) == 0 {
```

Comment:
If currentFreight were non-nil, the length of the stack couldn't possibly be 0.

```suggestion
	if currentFreight := status.FreightHistory.Current(); currentFreight == nil {
```

Reply:
From a pure Go perspective, it theoretically can if we have a bug elsewhere. This ensures that doesn't cause more problems.